To: vim_dev@googlegroups.com
Subject: Patch 7.3.219
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.219
Problem:    Can't compile with GTK on Mac.
Solution:   Add some #ifdef trickery. (Ben Schmidt)
Files:      src/os_mac_conv.c, src/os_macosx.m, src/vim.h
 

*** ../mercurial/vim73/src/os_mac_conv.c	2010-08-16 21:23:30.000000000 +0200
--- src/os_mac_conv.c	2011-06-13 01:55:17.000000000 +0200
***************
*** 14,19 ****
--- 14,21 ----
   */
  
  #define NO_X11_INCLUDES
+ #define BalloonEval int   /* used in header files */
+ 
  #include "vim.h"
  #ifndef FEAT_GUI_MAC
  # include <CoreServices/CoreServices.h>
*** ../mercurial/vim73/src/os_macosx.m	2010-10-10 17:08:28.000000000 +0200
--- src/os_macosx.m	2011-06-13 01:58:47.000000000 +0200
***************
*** 18,23 ****
--- 18,24 ----
  /* Avoid a conflict for the definition of Boolean between Mac header files and
   * X11 header files. */
  #define NO_X11_INCLUDES
+ #define BalloonEval int   /* used in header files */
  
  #include "vim.h"
  #import <Cocoa/Cocoa.h>
***************
*** 27,35 ****
   * Clipboard support for the console.
   * Don't include this when building the GUI version, the functions in
   * gui_mac.c are used then.  TODO: remove those instead?
!  * But for MacVim we need these ones.
   */
! #if defined(FEAT_CLIPBOARD) && (!defined(FEAT_GUI) || defined(FEAT_GUI_MACVIM))
  
  /* Used to identify clipboard data copied from Vim. */
  
--- 28,36 ----
   * Clipboard support for the console.
   * Don't include this when building the GUI version, the functions in
   * gui_mac.c are used then.  TODO: remove those instead?
!  * But for MacVim we do need these ones.
   */
! #if defined(FEAT_CLIPBOARD) && (!defined(FEAT_GUI_ENABLED) || defined(FEAT_GUI_MACVIM))
  
  /* Used to identify clipboard data copied from Vim. */
  
*** ../mercurial/vim73/src/vim.h	2011-05-19 18:26:34.000000000 +0200
--- src/vim.h	2011-06-13 01:57:17.000000000 +0200
***************
*** 121,126 ****
--- 121,127 ----
      || defined(FEAT_GUI_W32) \
      || defined(FEAT_GUI_W16) \
      || defined(FEAT_GUI_PHOTON)
+ # define FEAT_GUI_ENABLED  /* also defined with NO_X11_INCLUDES */
  # if !defined(FEAT_GUI) && !defined(NO_X11_INCLUDES)
  #  define FEAT_GUI
  # endif
*** ../vim-7.3.218/src/version.c	2011-06-13 01:32:42.000000000 +0200
--- src/version.c	2011-06-13 02:03:02.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     219,
  /**/

-- 
If Microsoft would build a car...
... the oil, water temperature, and alternator warning lights would
all be replaced by a single "General Protection Fault" warning light.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
